manifests/plugin/remote_execution.pp

```diff
@@ -1,7 +1,70 @@
+# = Foreman Remote Execution
 # Installs foreman_remote_execution plugin
-class foreman::plugin::remote_execution {
+#
+# === Parameters:
+#
+# $cockpit:: Install cockpit support
+#
+class foreman::plugin::remote_execution (
+  Boolean $cockpit = $::foreman::plugin::remote_execution::params::cockpit,
+) inherits foreman::plugin::remote_execution::params {
   include ::foreman::plugin::tasks
 
   foreman::plugin {'remote_execution':
   }
+
+  if $cockpit {
+    case $::osfamily {
+      'RedHat': {
+        case $::operatingsystem {
+          'fedora': {
+            $cockpit_package = 'rubygem-foreman_remote_execution-cockpit'
+          }
+          default: {
+            $cockpit_package = 'tfm-rubygem-foreman_remote_execution-cockpit'
+          }
+        }
+      }
+      default: {
+        fail("${::hostname}: foreman_remote_execution cockpit integration does not support osfamily ${::osfamily}")
+      }
+    }
+
+    package { $cockpit_package:
+      ensure => present,
+    }
+    ->
+    service { 'foreman-cockpit':
+      ensure     => running,
+      enable     => true,
+      hasstatus  => true,
```

Comment:
This defaults to true since puppet 2.7 and can be dropped:
```suggestion
```

Reply:
Updated